src/sys/mman.rs

```diff
@@ -603,3 +603,64 @@ pub fn shm_unlink<P: ?Sized + NixPath>(name: &P) -> Result<()> {
 
     Errno::result(ret).map(drop)
 }
+
+/// Creates and returns a new, or returns an existing, System V shared memory
+/// segment identifier.
+///
+/// For more information, see [`shmget(2)`].
+///
+/// [`shmget(2)`]: https://man7.org/linux/man-pages/man2/shmget.2.html
+pub fn shmget(key: key_t, size: size_t, shmflg: c_int) -> Result<c_int> {
```

Comment:
At the very least the result type should be a newtype pattern. I.e:

```rust
-> Result<ShmKey>
```

where 

```rust
#[derive(Debug, PartialEq, Eq, Hash)]
struct ShmKey(c_int)

impl ShmKey {
    unsafe fn to_raw(&self) -> c_int {
        self.0
    }
    // unsafe from_raw, other methods
}
```

This helps ensure that you will get sensible compiler errors.